aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -768,25 +769,25 @@ DimVector default_alldims(const Tensor& self, at::OptionalIntArrayRef dim_opt) {
 Tensor fft_fftshift(const Tensor& x, at::OptionalIntArrayRef dim_opt) {
   auto dim = default_alldims(x, dim_opt);
 
-  IntArrayRef x_sizes = x.sizes();
-  DimVector shift(dim.size());
+  SymIntArrayRef x_sizes = x.sym_sizes();
+  SymDimVector shift(dim.size());
   for (const auto i : c10::irange(dim.size())) {
     shift[i] = x_sizes[dim[i]] / 2;
   }
 
-  return at::roll(x, shift, dim);
+  return at::roll_symint(x, shift, dim);
```

Comment:
`roll_symint` is needed here: `x_sizes` is now symbolic, so `shift` has to be symbolic as well.